### ISSUE TITLE
when requesting xml, response is a truncated json response

### ISSUE THREAD

**Issue by pdumais** [freeconf/restconf#31]
When doing a POST with content-type "application/yang-data+json", the restconf server responds with a correctly formated response that conforms to our yang model.

But if the request is made with application/yang-data+xml. then the response is send as application/yang-data+json, but with the top level attribute stripped out.

**Comment by dhubler** [comment]
FreeCONF doesn't support XML
   https://freeconf.org/docs/reference/compliance/rfcs/
Curious, is this just for compliance? My impression was there is little
practical reason to support this, only theoretical.  I'm not sure how hard
this would be to support.

If you pass anything but application/yang-data+json you get regular JSON
and not namespaces per RFC7951 much like how gnmi supports both

https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#231-json-and-json_ietf

**Comment by pdumais** [comment]
>> If you pass anything but application/yang-data+json you get regular JSON
This means that issue #29 should not get resolved then. I'll take a look at the RFC. Do you remember where you saw that?

**Comment by pdumais** [comment]
actually, let me rephrase the issue.

When a request is made with content-type yang-data+xml, it is ok to respond with yang-data+json.
When when receiving the response, I expect it to be the same regardless. In our case, we are missing the top level node of the json document.

**Comment by dhubler** [comment]
Today we have:

When
  Accept: yang-data+json
Expect
  JSON IETF7951

When
   Accept: anything else
Get
  JSON


You want:

When
  Accept: yang-data+json or yang-data+xml
Expect
  JSON IETF7951

When
   Accept: anything else
Get
  JSON


If so, I kinda feel if the ask is for xml, we should error, not return JSON
(IETF7951 or otherwise)
